utils/src/array/u64.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{Array, SizedSerialize};
+use std::{
+    cmp::{Ord, PartialOrd},
+    fmt::{Debug, Display},
+    hash::Hash,
+    ops::Deref,
+};
+use thiserror::Error;
+
+// Errors returned by `U64` functions.
+#[derive(Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("invalid u64 length")]
+    InvalidU64Length,
```

Comment:
nit: Given this is the only item in here, I think we can just do `InvalidLength`